projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -467,21 +453,7 @@ private[projection] class R2dbcOffsetStore(
         } else
           newState
 
-      val statement = conn.createStatement(insertTimestampOffsetSql)
-
-      val offsetInserts =
-        if (filteredRecords.size == 1) {
-          val boundStatement = bindRecord(statement, filteredRecords.head)
-          R2dbcExecutor.updateOneInTx(boundStatement)
-        } else {
-          // TODO Try Batch without bind parameters for better performance. Risk of sql injection for these parameters is low.
-          val boundStatement =
-            filteredRecords.foldLeft(statement) { (stmt, rec) =>
-              stmt.add()
-              bindRecord(stmt, rec)
-            }
-          R2dbcExecutor.updateBatchInTx(boundStatement)
-        }
+      val offsetInserts = insertTimestampOffsetInTx(conn, filteredRecords)
```

Comment:
extracted this part into `def insertTimestampOffsetInTx` because same is needed from the `managementSetOffset`